src/java.base/share/classes/sun/security/x509/OtherName.java

```diff
@@ -89,8 +89,11 @@ public OtherName(DerValue derValue) throws IOException {
         DerInputStream in = derValue.toDerInputStream();
 
         oid = in.getOID();
-        DerValue val = in.getDerValue();
-        nameValue = val.toByteArray();
+        DerValue derValue1 = in.getDerValue();
+        if (!derValue1.isContextSpecific((byte)0) || !derValue1.isConstructed()) {
```

Comment:
It might be purely a matter of taste, but isn't `!(isCSTag0 && isConst)` easier to read?

Reply:
I have difficulty describing `!(a && b)`. There is no parentheses in human language and '!' has higher order than '&&'.

I thought about completely reverse the block but that means everything after the throw will be inside a block and I don't want to move so many lines.

Reply:
My wording for the &&: If the tag is not a constructed and context-specific tag number 0, then an exception is thrown. The parens denote that both conditions need to apply:
```
!(isCSTag0 && isConst)

true, true = !(true && true) = !true = false
true, false = !(true && false) = !false = true
false, true = !(false && true) = !false = true
false, false = !(false && false) = !false = true


!isCSTag0 || !isConst

true, true = !true || !true = false || false = false
true, false = !true || !false = false || true = true
false, true = !false || !true = true || false = true
false, false = !false || !false = true || true = true
```

Reply:
If you really like it, I'll write
```
        if (derValue1.isContextSpecific((byte) 0) && derValue1.isConstructed()) {
            nameValue = derValue1.data.toByteArray();
            gni = getGNI(oid, nameValue);
            if (gni != null) {
                name = gni.toString();
            } else {
                name = "Unrecognized ObjectIdentifier: " + oid.toString();
            }
        } else {
            throw new IOException("value is not [0]");
        }
```